### PR TITLE
warn for not using local stacks node with the signer

### PR DESCRIPTION
**Pull request by ASuciuX**
Current Status: Waiting for FAQ section to add link to.

- fixes https://github.com/stacks-network/stacks-core/issues/4730

Example display of message. Do we want to place it on only one line? Maybe make the message shorter.

image

### CODE REVIEW

File: stacks-signer/src/main.rs [hunk 1]
```diff
@@ -81,6 +81,13 @@ fn write_chunk_to_stdout(chunk_opt: Option<Vec<u8>>) {
 fn spawn_running_signer(path: &PathBuf) -> SpawnedSigner {
     let config = GlobalConfig::try_from(path).unwrap();
     let endpoint = config.endpoint;
+    // TODO: check if config.node_host is in a given list of possible local hosts and only if it now, then display the message?
+    warn!(
```
Comment (jferrant) [inline stacks-signer/src/main.rs:85]: I do like the idea of printing this if the config.node_host is not a localhost for sure. Makes it cleaner and less likely someone will ignore it :D Thanks for doing this!

Comment (ASuciuX) [reply stacks-signer/src/main.rs:85]: The `regtest-env` setup sees it as `stacks-node:20443`. Should I include `stacks-node` or other names on the checker besides `localhost`?

**Comment by hstove** [comment]
I think it could potentially cause more harm (or at least worry) than good if we're being strict about this warning with non-localhost setups. IMO it's probably best practice to have these two services on separate colocated machines, or at most using something like docker-compose, but we wouldn't really be able to check that. The main risk is that the signer isn't a registered event_observer, and that's easy to mess up if you're using an external node, but we don't have a good way of verifying that the signer is an event observer.

**Comment by jferrant** [comment]
> I think it could potentially cause more harm (or at least worry) than good if we're being strict about this warning with non-localhost setups. IMO it's probably best practice to have these two services on separate colocated machines, or at most using something like docker-compose, but we wouldn't really be able to check that. The main risk is that the signer isn't a registered event_observer, and that's easy to mess up if you're using an external node, but we don't have a good way of verifying that the signer is an event observer.

Well the warning is mostly in response to the audit as we are susecptible to man in the middle type situations.

File: stacks-signer/src/main.rs (hunk 1 above)
Comment (jferrant) [reply stacks-signer/src/main.rs:85]: Hm I don't like the idea of special casing too much around names beyond local host...perhaps instead just print the warning at all times. We can always revisit this later.

**Comment by ASuciuX** [comment]
Updated message displayed after having all of it as 1 line

image

File: stacks-signer/src/main.rs [hunk 2]
```diff
@@ -81,6 +81,12 @@ fn write_chunk_to_stdout(chunk_opt: Option<Vec<u8>>) {
 fn spawn_running_signer(path: &PathBuf) -> SpawnedSigner {
     let config = GlobalConfig::try_from(path).unwrap();
     let endpoint = config.endpoint;
+    warn!(
+        "The signer is primarily designed for use with a local stacks node. \
```
Comment (ASuciuX) [inline stacks-signer/src/main.rs]: @jferrant @hstove would you suggest any changes to the message? Also if we'll have a link for faq in the future, I could add that. Thanks

File: stacks-signer/src/main.rs [hunk 3]
```diff
@@ -81,6 +81,12 @@ fn write_chunk_to_stdout(chunk_opt: Option<Vec<u8>>) {
 fn spawn_running_signer(path: &PathBuf) -> SpawnedSigner {
     let config = GlobalConfig::try_from(path).unwrap();
     let endpoint = config.endpoint;
+    warn!(
+        "Reminder: The signer is primarily designed for use with a local or subnet network stacks node. \
```
Comment (ASuciuX) [inline stacks-signer/src/main.rs:85]: @jferrant @hstove would you suggest any changes to the message? Also if we'll have a link for faq in the future, I could add that. Thanks

**Review by hstove — APPROVED** [review]

**Review by jferrant — APPROVED** [review]

**Pull request merged**